Add FAQ answer about API stability & backwards compatibility (#9855)
pydata/xarray: doc/getting-started-guide/faq.rst

                 url    = {https://doi.org/10.5281/zenodo.59499}
                }

.. _api stability:

How stable is Xarray's API?
---------------------------

Xarray tries very hard to maintain backwards compatibility in our :ref:`api` between released versions.
Whilst we do occasionally make breaking changes in order to improve the library,

> we try to `signpost changes <https://docs.xarray.dev/en/stable/contributing.html#backwards-compatibility>`_ with ``DeprecationWarnings`` for many months in advance.

Reviewer: ```suggestion
we `signpost changes <https://docs.xarray.dev/en/stable/contributing.html#backwards-compatibility>`_ with ``DeprecationWarnings`` for many releases in advance.
```

Reviewer: Or

```suggestion
our policy is to `signpost changes <https://docs.xarray.dev/en/stable/contributing.html#backwards-compatibility>`_ with ``DeprecationWarnings`` for many releases in advance.
```
